### ISSUE TITLE
Unexpected error, when second argument of isinstance function is Union Type

### ISSUE THREAD

**Issue by Alexxxir** [python/mypy#12363]
**Bug Report**

Unexpected error, when second argument of isinstance function is Union Type

**To Reproduce**

```python
A = int | str

def a(b) -> None:
    if isinstance(b, A):
        print(b)

```

**Expected Behavior**

The code should typecheck without errors.

**Actual Behavior**

```bash
test.py:4: error: Parameterized generics cannot be used with class or instance checks
test.py:4: error: Argument 2 to "isinstance" has incompatible type "object"; expected "Union[type, UnionType, Tuple[Union[type, UnionType, Tuple[Any, ...]], ...]]"
```

**Your Environment**

- Mypy version used: 0.941
- Python version used: 3.10
- Operating system and version:

**Comment by KotlinIsland** [comment]
https://mypy-play.net/?mypy=latest&python=3.10&gist=4c0ace00d675e36f20200f5c12e631b5

**Comment by KotlinIsland** [comment]
Duplicate of #11673
